app/code/Magento/Ui/Config/Data.php

```diff
@@ -64,26 +65,37 @@ class Data implements \Magento\Framework\Config\DataInterface
      */
     private $argumentInterpreter;
 
+    /**
+     * Store manager.
+     *
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
     /**
      * @param string $componentName
      * @param ReaderFactory $readerFactory
      * @param CacheInterface $cache
      * @param SerializerInterface $serializer
      * @param InterpreterInterface $argumentInterpreter,
+     * @param StoreManagerInterface|null $storeManager
      */
     public function __construct(
         $componentName,
         ReaderFactory $readerFactory,
         CacheInterface $cache,
         SerializerInterface $serializer,
-        InterpreterInterface $argumentInterpreter
+        InterpreterInterface $argumentInterpreter,
+        StoreManagerInterface $storeManager = null
     ) {
         $this->readerFactory = $readerFactory;
         $this->cache = $cache;
         $this->serializer = $serializer;
         $this->componentName = $componentName;
         $this->argumentInterpreter = $argumentInterpreter;
-        $this->cacheId = static::CACHE_ID . '_' . $componentName;
+        $this->storeManager = $storeManager ?:
```

Comment:
No need to assign property which is only used in constructor.

Reply:
We set custom js files for widget, these settings cached for all stores and recently_viewed_widget tries to load custom JS files, which lead to 404 response (these JS files generated for other store)
**File:**
Custom_Theme/Magento_Catalog/ui_component/widget_recently_viewed.xml
**Content:**
<listing xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Ui:etc/ui_configuration.xsd">
    <columns name="widget_columns">
        <column name="status" sortOrder="10">
            <argument name="data" xsi:type="array">
                <item name="config" xsi:type="array">
                    <item name="displayArea" xsi:type="string">details-area</item>
                    <item name="bodyTmpl" xsi:type="string">YourVendor_CustomTheme/product/status</item>
                    <item name="component" xsi:type="string">**YourVendor_CustomTheme/js/product/status**</item>
                </item>
            </argument>
        </column>
    </columns>
</listing>

Reply:
@agorbulin please put this to "manual testing scenario" section.

Reply:
Done